src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -0,0 +1,156 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.metrics.MetricsFile;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Worker class to collect insert size metrics.
+ */
+public final class InsertSizeMetricsCollectorSpark implements Serializable{
+    private static final long serialVersionUID = 1L;
+
+    private final JavaRDD<Integer> lengthRDD;
```

Comment:
It was used on lines 81, 91, and 98. But if you think recomputing the RDD is a better tradeoff, I'll remove it and recompute when necessary.